os-autoinst-obs-auto-submit

```diff
@@ -18,6 +18,16 @@ XMLSTARLET=$(command -v xmlstarlet || true)
 # shellcheck source=/dev/null
 . "$(dirname "$0")"/_common
 
+# extract :Update
+update_prj="${submit_target_extra##*:}"
+# extract openSUSE:Backports:SLE-15-SP6
+current_prj="${submit_target_extra%:*}"
+# extact the actual version to compare
+current_prj="${current_prj%:*}"
+# awk iterates through all the fields in the match to avoid fixed position
+latest_submit_target_extra_version=$(osc dists | grep "$current_prj" | awk -v prj="$current_prj" '{for (i=1; i<=NF; i++) if ($i ~ "^" prj ":") print $i}' | sort --version-sort | tail -n 1)
+submit_target_extra="$latest_submit_target_extra_version:$update_prj"
```

Comment:
This is not very straight-forward to read and feels like it will break easily. Any chance we can unit test this with a couple of examples?

Reply:
This can break in many ways and it depends on the $submit_target_extra
I tried to generalize it but at the end the main goal is to have an additional SR to the Backports. There could be some checks or some mapping (some things which I considered when I wrote the code)... lets make it work and then see where it can improve?!

Reply:
note that this still doesnt work

Reply:
As you seem to agree with me, are you planning to add a test for this? 🤔

Reply:
Honestly I havent thought about that. There is no test in the repo for it and I am not sure how easy will be to test against OBS. 
When I said `There could be some checks` I meant some conditions in the script to catch some cases.

Reply:
But if you insists I will try to add one or two assertions for the variable. sounds good? (what I dont know is how urgent is this, and whether can be another PR, in order to move on with this one. This is already some time we try to complete the task and I would like to see some progress)

Reply:
The fact that we have no tests right now is not a very helpful argument to my mind. Even writing this code myself I would prepare tests first to confirm what it does.

I'm not talking about testing against production infra btw. Something like a few examples of possible `osc dists` with one or two examples of a project config we use.

Reply:
@kalikiana I am sure this will start a new iteration of changes. But I did add a test but... 
not sure you will agree with the approach and I dont know if those lines should be separate function in the script. And Furthermore I dont think it covers all you expect to see. I have done assumptions for now. and I couldnt think how the script would look like in the near future. so it is kinda poor approach🤞🏾